plugins/http_plugin/include/eosio/http_plugin/beast_http_session.hpp

```diff
@@ -0,0 +1,577 @@
+#pragma once
+
+#include <eosio/http_plugin/common.hpp>
+
+#include <algorithm>
+#include <cstdlib>
+#include <functional>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <thread>
+#include <vector>
+#include <sstream>
+
+extern fc::logger logger;
```

Comment:
This is rather fragile. net_plugin also has a `logger` that is file scope. It is in `eosio` namespace, but others might not be. Especially if someone has an external plugin. Instead, please pass the logger into the constructor and move `fail()` to be member function.

Reply:
Would it be bad form to put it into plugin_state?  Also it will probably need to be shared_ptr if it is no longer global

Reply:
Aso there a couple instances of just calling ilog, elog, etc. without reference to the logger  .. should all logging go through the logger instance ?